index.js

```diff
@@ -46,7 +85,17 @@ function requestAsEventEmitter(opts) {
 			fn = electron.net || electron.remote.net;
 		}
 
+		let progressInterval;
+
 		const req = fn.request(opts, res => {
+			clearInterval(progressInterval);
+
+			ee.emit('uploadProgress', {
+				percent: 1,
```

Comment:
> Why is `percent` set to 1? Probably missing something here :).

It's calculating `percent` to be from `0` to `1`. So `1` means it's the last upload progress event and it's done.

> I'm also wondering why it is emitted?

Initially it was only for convenience. But there are also cases where `total` can be unavailable (like you said, when streaming), so got makes sure that `percent` is always available.